src/bundle/Controller/LocationController.php

```diff
@@ -8,25 +8,47 @@
  */
 namespace EzSystems\HybridPlatformUiBundle\Controller;
 
+use eZ\Publish\API\Repository\ContentService;
 use eZ\Publish\API\Repository\Values\Content\Content;
 use eZ\Publish\Core\MVC\Symfony\View\ContentView;
 use EzSystems\HybridPlatformUi\Form\UiFormFactory;
 use EzSystems\HybridPlatformUi\Repository\UiLocationService;
+use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
 
 class LocationController extends TabController
 {
-    public function contentViewTabAction(
-        ContentView $view,
+    /**
+     * @var UiLocationService
+     */
+    private $uiLocationService;
+
+    /**
+     * @var UiFormFactory
+     */
+    private $formFactory;
+
+    public function __construct(
         UiLocationService $uiLocationService,
+        RouterInterface $router,
+        ContentService $contentService,
```

Comment:
Probably one for a different PR but I'm wondering if `TabController` can be refactored to remove the dependency on ContentService? Seems strange that we need `ContentService` in this controller especially when resetLocation isn't even called